src/js/common/Button.spec.jsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import userEvent from "@testing-library/user-event";
+import { screen, render as rtlRender } from "@testing-library/react";
+
+import Button from "./Button";
+
+const render = ({ buttonText, ...props }) =>
+    rtlRender(<Button {...props}>{buttonText}</Button>);
+
+describe("<Button />", () => {
+    const buttonText = "Button";
+
+    it("renders as expected", () => {
+        render({ buttonText });
+
+        expect(screen.getByRole("button")).toBeInTheDocument();
+        expect(screen.getByText(buttonText)).toBeInTheDocument();
+    });
+
+    it("is clickable", () => {
```

Comment:
Open question here: Do we think providing these two tests (button functionality) would be giving too much to the candidate? Or are we using these as a "Hey, this is the kind of test we want to see, go build more"?

Reply:
The coding challenge isn't a way to stump a candidate, it's to see what they do with what we provide them. If we don't provide them with all the tools, we're setting them up for failure. We count adding tests to the challenge as a +1, but that approach is biased to 2 sorts of people: a person who already knows how to write unit and integration tests and/or a developer who has the time to look up how to write unit and integration tests in the span of them having this challange.

Also, the only mention of writing tests in the rubric is simply asking "should you write unit/integration tests", not "you should be writing tests for all of the code you provide us". Given that approach, it's up to the candidate how they would want to proceed, do they look through the codebase for similar examples of tests, do they not like the rudimentary tests provided, can the candidate write more verbose tests.

Reply:
At least in my mind giving them _just_ the test for "renders as expected" seems like a good launching point. If they ended up coming up with more tests like these for testing states of the buttons, etc. it would show they have a good grasp of how and what to test. That's just my two cents though, take it with a grain of salt.

Reply:
This is not a discussion for a PR, I'd be glad to discuss this via video